.pre-commit-config.yaml

```diff
@@ -0,0 +1,46 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+
+# Built-in hooks
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: "v4.4.0"
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+        args: ['--maxkb=500000'] # 500MB
+
+# clang-format
+-   repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: "v16.0.6"
+    hooks:
+    -   id: clang-format
+        args: ["--style=Google"]
```

Comment:
I feel like if we're hoping to upstream some of the stuff in here, we may want to make it --style=LLVM? Or just have this default to using a .clang-format config file in the project root (see https://github.com/google/heir/pull/89 for a proposed initial config)

Reply:
Right! I wanted to but copybara doesn't apply custom clangtidy rules (that I know of) so this might cause a breakage in the exported files.

Reply:
Let's follow up with this on your #89